damnit/ctxsupport/ctxrunner.py

```diff
@@ -211,16 +212,28 @@ def __init__(self, data, ctx):
     def create(cls, ctx_file: ContextFile, inputs, run_number, proposal):
         res = {'start_time': np.asarray(get_start_time(inputs['run_data']))}
 
+        def get_dep_or_default(var, arg_name, dep_name):
+            """
+            Helper function to get either the value returned from the dependency
+            `dep_name` of `var`, if any, or the default value of the argument in
+            the function signature.
+            """
+            value = res.get(dep_name)
+            if value is None:
+                value = inspect.signature(var.func).parameters[arg_name].default
+
+            return value
+
         for name in ctx_file.ordered_vars():
             var = ctx_file.vars[name]
 
             try:
                 # Add all variable dependencies
-                kwargs = { arg_name: res.get(dep_name)
+                kwargs = { arg_name: get_dep_or_default(var, arg_name, dep_name)
                            for arg_name, dep_name in var.arg_dependencies().items() }
 
                 # If any are None, skip this variable since we're missing a dependency
```

Comment:
```suggestion
                # Check for missing dependencies with no default value in the function
```

Since we're no longer looking for None. :slightly_smiling_face:

Reply:
Hah, indeed :) Fixed in 6085018.